### PR TITLE
Add build tags to support both x86 and ARM compilation on macOS

### PR DESCRIPTION
**Pull request by Spaider**
Fix for #1068 

**Changes**

* Introduce `arm64` and `amd64` build tags
* Appropriate changes in documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -288,13 +288,13 @@ brew upgrade icu4c
 To compile for macOS X on x86:
 
 ```bash
-go build --tags "darwin,amd64"
```
Comment (rittneje) [inline README.md]: These instructions were actually totally wrong. It should be using `GOOS` and `GOARCH` for this. The same applies for Windows and Linux. (Also the flag is `-tags` not `--tags`.)

File: README.md [hunk 2]
```diff
@@ -40,7 +40,7 @@ This package follows the official [Golang Release Policy](https://golang.org/doc
     - [Alpine](#alpine)
     - [Fedora](#fedora)
     - [Ubuntu](#ubuntu)
-  - [Mac OSX](#mac-osx)
+  - [macOS X](#mac-osx)
```
Comment (rittneje) [inline README.md:43]: In modern times it is simply named `macOS`.

**Comment by rittneje** [comment]
@mattn Do you know if GitHub has any M1/M2 build agents we could add to the workflow? https://github.com/mattn/go-sqlite3/blob/master/.github/workflows/go.yaml

**Comment by mattn** [comment]
No I don't.

**Comment by Spaider** [comment]
> @mattn Do you know if GitHub has any M1/M2 build agents

[Issue](https://github.com/actions/virtual-environments/issues/2187) is still open. Looks like it's in plans but not there yet.

File: README.md (hunk 1 above)
Comment (Spaider) [reply README.md]: Sorry, I'm not following where's my fault here. I've added architecture flags exactly in a way it was for OS selection: by explicitly define target OS/arch. I'm a roocky in cgo and not sure how to re-write this to make it refer to `GOOS` and `GOARCH` env vars: `#cgo` flag does not have `if-then-else` notion.

Comment (rittneje) [reply README.md]: The original instructions (on master) were incorrect.

If you are doing a _native_ compile, it is just `go build`.
If you are doing a _cross_ compile, it is `GOOS=<target-os> GOARCH=<target-arch> CGO_ENABLED=1 CC=<c-cross-compiler> go build`
For example: `GOOS=darwin GOARCH=amd64 CGO_ENABLED=1 CC=o64-clang go build`

Separately from that, if you are trying to include optional features via build tags, in modern versions of Go you provide a comma-separated list via `-tags`.
> -tags tag,list
	a comma-separated list of build tags to consider satisfied during the
	build. For more information about build tags, see the description of
	build constraints in the documentation for the go/build package.
	(Earlier versions of Go used a space-separated list, and that form
	is deprecated but still recognized.)

For example: `go build -tags sqlite_stat4,secure_delete` 

Note that all of this is separate from the `#cgo` directive changes you made. (Those changes are correct.)

Comment (Spaider) [reply README.md]: Thanks for an explanation, now it makes perfect sense. Looks like the problem of both _native_ and _cross-compilcation_ can be resolved only by external tool. For example, shell script which will set `CGO_CGOCFLAGS` and `CGO_LDFLAGS` based on _target_ `GOOS` and `GOARCH`. But this is beyond the scope of my PR.

**Review by mattn — APPROVED** [review]

File: sqlite3_opt_icu.go [hunk 3]
```diff
@@ -10,8 +10,10 @@ package sqlite3
 /*
 #cgo LDFLAGS: -licuuc -licui18n
 #cgo CFLAGS: -DSQLITE_ENABLE_ICU
-#cgo darwin CFLAGS: -I/usr/local/opt/icu4c/include
-#cgo darwin LDFLAGS: -L/usr/local/opt/icu4c/lib
+#cgo darwin,amd64 CFLAGS:  -I/usr/local/opt/icu4c/include
+#cgo darwin,amd64 LDFLAGS: -L/usr/local/opt/icu4c/lib
+#cgo darwin,arm64 CFLAGS:  -I/opt/homebrew/opt/icu4c/include
+#cgo darwin,arm64 LDFLAGS: -L/opt/homebrew/opt/icu4c/lib
```
Comment (mattn) [inline sqlite3_opt_icu.go:16]: I'm wondering why this is different for them.
```
-I/usr/local/opt/icu4c/include -I/opt/homebrew/opt/icu4c/include
-L/usr/local/opt/icu4c/lib -L/opt/homebrew/opt/icu4c/lib
```
This is not good?

Comment (rittneje) [reply sqlite3_opt_icu.go:16]: @mattn Not sure I understand your question, but Homebrew installs to a different location on arm64 (i.e, M1/M2) than on amd64 (i.e., Intel), as mentioned in the docs. https://docs.brew.sh/Installation

We could also choose to include _both_ directories for macOS regardless of GOARCH, and I suppose that would also work. (Although I think we should put /opt/homebrew first in that case.)

**Comment by zzet** [comment]
Hey folks! I'm wondering if any progress there?

**Comment by Spaider** [comment]
I've resolved the conflict in `README.md`.

File: sqlite3_opt_icu.go (hunk 3 above)
Comment (ARolek) [reply sqlite3_opt_icu.go:16]: I just hit this issue on a M1 Mac and these locations are correct. This PR along with the build tags `go build -tags "libsqlite3 darwin arm64"` allowed me to build against the sqlite version installed via homebrew.

**Comment by ARolek** [comment]
Just checking in. It looks like this PR was approved, but never merged. Is there anything blocking the merge at this time?

**Pull request merged**